api/admin/controller/__init__.py

```diff
@@ -1650,6 +1650,81 @@ def check_name_unique(self, new_service, name):
             # changes to an existing service unless you've also changed its name.
             return INTEGRATION_NAME_ALREADY_IN_USE
 
+    @classmethod
+    def url_variants(cls, url, check_protocol_variant=True):
+        """Generate minor variants of a URL -- HTTP vs HTTPS, trailing slash
```

Comment:
It's okay for now because `discovery_services` calls `validate_formats` before calling this, but in general: it might be worth putting the url through `Validator._is_url` right at the top.  (If the URL isn't valid, then there's no point going ahead with generating variants on it.)